src/System.Drawing.Design.Primitives/src/System/Drawing/Design/ToolboxItem.cs

```diff
@@ -0,0 +1,1100 @@
+        /// However, the loaded type is not added to the list of references in the designer host.
+        /// </summary>
+        public Type GetType(IDesignerHost host)
+        {
+            return GetType(host, AssemblyName, TypeName, false);
+        }
+
+        /// <summary>
+        /// This utility function can be used to load a type given a name.  AssemblyName and
+        ///     designer host can be null, but if they are present they will be used to help
+        ///     locate the type.  If reference is true, the given assembly name will be added
+        ///     to the designer host's set of references.
+        /// </summary>
+        protected virtual Type GetType(IDesignerHost host, AssemblyName assemblyName, string typeName, bool reference)
+        {
+            ITypeResolutionService ts = null;
+            Type type = null;
+
+            if (typeName == null)
+            {
+                throw new ArgumentNullException("typeName");
+            }
+
+            if (host != null)
+            {
+                ts = (ITypeResolutionService)host.GetService(typeof(ITypeResolutionService));
+            }
+
+            if (ts != null)
+            {
+                if (reference)
+                {
+                    if (assemblyName != null)
+                    {
+                        ts.ReferenceAssembly(assemblyName);
+                        type = ts.GetType(typeName);
+                    }
+                    else
+                    {
+                        // Just try loading the type.  If we succeed, then use this as the
+                        // reference.
+                        type = ts.GetType(typeName);
+                        if (type == null)
+                        {
+                            type = Type.GetType(typeName);
+                        }
+                        if (type != null)
+                        {
+                            ts.ReferenceAssembly(type.Assembly.GetName());
+                        }
+                    }
+                }
+                else
+                {
+                    if (assemblyName != null)
+                    {
+                        Assembly a = ts.GetAssembly(assemblyName);
+                        if (a != null)
+                        {
+                            type = a.GetType(typeName);
+                        }
+                    }
+
+                    if (type == null)
+                    {
+                        type = ts.GetType(typeName);
+                    }
+                }
+            }
+            else
+            {
+                if (!String.IsNullOrEmpty(typeName))
+                {
+                    if (assemblyName != null)
+                    {
+                        Assembly a = null;
+                        try
+                        {
+                            a = Assembly.Load(assemblyName);
+                        }
```

Comment:
@stephentoub what's the common pattern for assembly loading in corefx? Entirely swallowing potential exceptions or logging them out somewhere?